docs/asciidoc/static/private-gem-repo.asciidoc

```diff
@@ -0,0 +1,53 @@
+[[private-rubygem]]
+=== Private Gem Repositories
+
+The Logstash plugin manager connects to a Ruby gems repository to install and update Logstash plugins. By default, this
+repository is http://rubygems.org.
+
+Some use cases are unable to use the default repository, as in the following examples:
+
+* A firewall blocks access to the default repository.
+* You are developing your own plugins locally.
+* Airgap requirements on the local system.
+
+When you use a custom gem repository, be sure to make plugin dependencies available.
+
+Several open source projects enable you to run your own plugin server, among them:
+
+* https://github.com/geminabox/geminabox[Geminabox]
+* https://github.com/PierreRambaud/gemirro[Gemirro]
+* https://gemfury.com/[Gemfury]
+* http://www.jfrog.com/open-source/[Artifactory]
+
+==== Editing the Gemfile
+
+The gemfile is a configuration file that specifies information required for plugin management. Each gem file has a
+`source` line that specifies a location for plugin content.
+
+By default, the gemfile's `source` line reads:
+
+[source,shell]
+----------
+# This is a Logstash generated Gemfile.
+# If you modify this file manually all comments and formatting will be lost.
++
+source "https://rubygems.org"
```

Comment:
Typo -- that + sign above 'source' shouldn't be there. I'll fix it when I
incorporate the other feedback.

pe

On Thu, Nov 19, 2015 at 8:16 AM, Alvin Chen <notifications@github.com>
wrote:

> In docs/asciidoc/static/private-gem-repo.asciidoc
> <https://github.com/elastic/logstash/pull/4230#discussion_r45361560>:
>
> > +* https://gemfury.com/[Gemfury]
> > +* http://www.jfrog.com/open-source/[Artifactory]
> > +
> > +==== Editing the Gemfile
> > +
> > +The gemfile is a configuration file that specifies information required for plugin management. Each gem file has a
> > +`source` line that specifies a location for plugin content.
> > +
> > +By default, the gemfile's `source` line reads:
> > +
> > +[source,shell]
> > +----------
> > +# This is a Logstash generated Gemfile.
> > +# If you modify this file manually all comments and formatting will be lost.
> > ++
> > +source "https://rubygems.org"
>
> This line and the following instance don't seem to be rendering correctly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4230/files#r45361560>.
>